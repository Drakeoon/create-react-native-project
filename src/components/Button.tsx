import React from 'react';
import {
  StyleProp,
  // StyleSheet,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';

import Typography, { TypographyType, TypographyVariant } from './Typography';

type Props = {
  onPress: () => void;
  label: string;
  labelVariant?: TypographyVariant;
  labelType?: TypographyType;
  style?: StyleProp<ViewStyle>;
};

const Button = ({ onPress, label, labelType, labelVariant, style }: Props) => {
  return (
    <TouchableOpacity onPress={onPress} style={style}>
      <Typography variant={labelVariant} type={labelType}>
        {label}
      </Typography>
    </TouchableOpacity>
  );
};

export default React.memo(Button);

// const styles = StyleSheet.create({});
