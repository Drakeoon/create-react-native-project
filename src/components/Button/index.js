import PropTypes from 'prop-types'
import React from 'react'
import colorPackage from 'color'
import { BaseButton } from 'react-native-gesture-handler'
import { Platform, TouchableHighlight } from 'react-native'

import { fontColor } from 'src/styles/colors'

const ANDROID_VERSION_LOLLIPOP = 21
const isAndroid = Platform.OS === 'android'
const rippleSupported = isAndroid && Platform.Version >= ANDROID_VERSION_LOLLIPOP
const Touchable = rippleSupported ? BaseButton : TouchableHighlight

export default class Button extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    color: PropTypes.string,
    disabled: PropTypes.bool,
    onPress: PropTypes.func.isRequired,
    style: PropTypes.any,
  }

  render() {
    const { style, color, disabled, onPress, children } = this.props
    const rippleColor = style && 'backgroundColor' in style && isAndroid ? fontColor : color

    const calculatedRippleColor = colorPackage(rippleColor || fontColor)
      .alpha(0.2)
      .rgb()
      .string()

    return (
      <Touchable
        activeOpacity={0.4}
        disabled={disabled}
        onPress={onPress}
        rippleColor={calculatedRippleColor}
        style={style}
        underlayColor={colorPackage(calculatedRippleColor)
          .fade(0.5)
          .rgb()
          .string()}
      >
        {children}
      </Touchable>
    )
  }
}
