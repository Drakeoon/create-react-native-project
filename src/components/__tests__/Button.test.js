import React from 'react'
import renderer from 'react-test-renderer'
import { Text } from 'react-native'

import Button from '../Button'

describe('<Button />', () => {
  const defaultProps = {
    onPress: jest.fn(),
  }
  const wrapper = renderer.create(
    <Button {...defaultProps}>
      <Text>Button</Text>
    </Button>,
  )

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
