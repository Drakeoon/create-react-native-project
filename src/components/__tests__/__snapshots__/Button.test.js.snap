// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <Text
    style={null}
  >
    Button
  </Text>
</View>
`;
